### DISCUSSION TITLE
How to resize the images to a specific width with the unit of centimeter (Not Pixels)?

### DISCUSSION THREAD

**Discussion by LePingKYXK** [ImageMagick/ImageMagick#6397] [Help] [answered]
### ImageMagick version

7.1.0-Q16-HDRI

### Operating system, version and so on

Windows 10

### Description

I have read the online manual for resizing images via the command line (cmd in Windows). However, all the codes provided only the pixel times pixel, like 
```bash
magick dragon_sm.gif    -resize 64x64  resize_dragon.gif
```

Is it possible to specify the size of the width with the unit of centimeters? e.g.
```bash
magick dragon_sm.gif   -resize   "width 8cm"  resize_dragon.gif
```

I think this would be much more convenient for most people.

**Comment by snibgo** [comment, marked as answer]
If you want the width to be 8 cm, the simplest way is to set the density. This doesn't change the number of pixels.

```
magick in.png -units PixelsPerCentimeter -density %[fx:w/8] out.png
```

An alternative is to keep the density constant, and use `-resize` to change the number of pixels. For example, if you know the density is `D` PixelsPerCentimeter (where `D` is some number), then:
```
magick in.png -resize %[fx:D*8] out.png
```

**Reply by LePingKYXK** [reply]
Thank you very much for your kindly reply. The first code works well for me. 

I have further questions about the second code. Does the density mean the DPI?  I tried 
```bash
magick test.png -resize %[fx:300*8] test01.png
```
However, it did not change the width of the image, only reduced the volume. 
I am not sure if this is due to my misunderstanding of the `density`.

**Comment by snibgo** [comment]
> Does the density mean the DPI? 

Yes.

> However, it did not change the width of the image, only reduced the volume.

By "volume", do you mean the number of bytes in the file? Even when the pixels don't change, the file size may change due to using different compression parameters.

If the width was already `300*8` pixels, the `-resize` will do nothing.

Does that answer the question? If not, then please supply a sample input file.

**Reply by LePingKYXK** [reply]
Thank you for your patient instruction. Your answers are very clear and helpful.
